mkosi/__init__.py

```diff
@@ -385,9 +386,68 @@ def configure_os_release(context: Context) -> None:
 
             newosrelease.rename(osrelease)
 
+        osrelease_content = read_env_file(osrelease)
+
         if ArtifactOutput.os_release in context.config.split_artifacts:
             shutil.copy(osrelease, context.staging / context.config.output_split_os_release)
 
+    # https://www.freedesktop.org/software/appstream/docs/sect-Metadata-OS.html
+    osid = context.config.appstream_id or osrelease_content.get("ID", "linux")
+    name = context.config.appstream_name or osrelease_content.get("NAME") or context.config.image_id or osid
+    if context.config.appstream_description:
+        description = context.config.appstream_description
+    elif "PRETTY_NAME" in osrelease_content:
+        description = f"{osrelease_content.get("PRETTY_NAME")} built with mkosi"
+    else:
+        description = "Image built with mkosi"
+    icon = context.config.appstream_icon or "https://brand.systemd.io/assets/svg/systemd-logomark.svg"
+    if context.config.appstream_homepage:
+        home_url = context.config.appstream_homepage
+    elif "HOME_URL" in osrelease_content:
+        home_url = osrelease_content.get("HOME_URL")
+    else:
+        home_url = None
+    if home_url:
+        url = urlparse(home_url)
+        if not url.netloc:
+            home_url = None
+    if context.config.appstream_id:
+        id = context.config.appstream_id
+    elif home_url:
+        netloc = url.netloc.split(".")
+        netloc.reverse()
+        netloc.remove("www")
+        id = ".".join(netloc)
+        id += f".{osid}"
```

Comment:
Because it got marked as outdated due to refactorings:

> But it is a valid subdomain (e.g. distro.www.company.org). Not sure who would do that stuff but there is no reason to not just do the correct thing.

```suggestion
        netloc = url.netloc.removeprefix("www.").split(".")
        id = ".".join(reversed(netloc))
        id += f".{osid}"
```

Reply:
there is no distribution doing anything like that

Reply:
But there is no reason to knowingly implement the wrong behavior instead of just doing it the right way (tm)